adapters/adnuntius/adnuntius_utils.go

```diff
@@ -0,0 +1,202 @@
+package adnuntius
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"strconv"
+
+	"github.com/prebid/openrtb/v20/openrtb2"
+	"github.com/prebid/prebid-server/v3/openrtb_ext"
+	"github.com/prebid/prebid-server/v3/util/jsonutil"
+)
+
+func setHeaders(ortbRequest openrtb2.BidRequest) http.Header {
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	if ortbRequest.Device != nil {
+		if ortbRequest.Device.IP != "" {
+			headers.Add("X-Forwarded-For", ortbRequest.Device.IP)
+		}
+		if ortbRequest.Device.UA != "" {
+			headers.Add("user-agent", ortbRequest.Device.UA)
+		}
+	}
+	return headers
+}
+
+func makeEndpointUrl(ortbRequest openrtb2.BidRequest, a *adapter, noCookies bool) (string, []error) {
+	uri, err := url.Parse(a.endpoint)
+	endpointUrl := a.endpoint
+	if err != nil {
+		return "", []error{fmt.Errorf("failed to parse Adnuntius endpoint: %v", err)}
+	}
+
+	gdpr, consent, err := getGDPR(&ortbRequest)
+	if err != nil {
+		return "", []error{fmt.Errorf("failed to parse Adnuntius endpoint: %v", err)}
+	}
+
+	if !noCookies {
+		var deviceExt extDeviceAdnuntius
+		if ortbRequest.Device != nil && ortbRequest.Device.Ext != nil {
+			if err := jsonutil.Unmarshal(ortbRequest.Device.Ext, &deviceExt); err != nil {
+				return "", []error{fmt.Errorf("failed to parse Adnuntius endpoint: %v", err)}
+			}
+		}
+
+		if deviceExt.NoCookies {
+			noCookies = true
+		}
+	}
+
+	_, offset := a.time.Now().Zone()
+	tzo := -offset / minutesInHour
+
+	q := uri.Query()
+	if gdpr != "" {
+		endpointUrl = a.extraInfo
+		q.Set("gdpr", gdpr)
+	}
+
+	if consent != "" {
+		q.Set("consentString", consent)
+	}
+
+	if noCookies {
+		q.Set("noCookies", "true")
+	}
+
+	q.Set("tzo", fmt.Sprint(tzo))
```

Comment:
The fmt.Sprint function is used to convert tzo to a string. It would be more efficient to use `strconv.Itoa` for this purpose.
```
	q.Set("tzo", strconv.Itoa(tzo))
```